File: ainpc-core-plugin/src/main/java/ro/ainpc/world/patch/PatchPlannerResult.java
package ro.ainpc.world.patch;

import java.util.List;

public record PatchPlannerResult(
    GapReport gapReport,
    List<PatchCandidate> candidates,
    List<PatchPlan> patchPlans,
    List<String> warnings,
    List<String> errors
) {
    public PatchPlannerResult {
        candidates = List.copyOf(candidates != null ? candidates : List.of());
        patchPlans = List.copyOf(patchPlans != null ? patchPlans : List.of());
        warnings = List.copyOf(warnings != null ? warnings : List.of());
        errors = List.copyOf(errors != null ? errors : List.of());
    }

    public boolean success() {
        return errors.isEmpty();
    }
}

File: ainpc-core-plugin/src/main/kotlin/ro/ainpc/world/patch/PatchPlannerResult.kt
package ro.ainpc.world.patch

class PatchPlannerResult(
    private val gapReport: GapReport?,
    candidates: List<PatchCandidate>?,
    patchPlans: List<PatchPlan>?,
    warnings: List<String>?,
    errors: List<String>?
) {
    private val candidates: List<PatchCandidate> = (candidates ?: emptyList()).toList()
    private val patchPlans: List<PatchPlan> = (patchPlans ?: emptyList()).toList()
    private val warnings: List<String> = (warnings ?: emptyList()).toList()
    private val errors: List<String> = (errors ?: emptyList()).toList()

    fun gapReport(): GapReport? = gapReport
    fun candidates(): List<PatchCandidate> = candidates
    fun patchPlans(): List<PatchPlan> = patchPlans
    fun warnings(): List<String> = warnings
    fun errors(): List<String> = errors

    fun success(): Boolean = errors.isEmpty()
}
